backend/inspirehep/pidstore/providers/texkey.py

```diff
@@ -124,42 +114,71 @@ def create(
     ):
         if pid_value:
             texkey = cls.create_or_update_texkey_in_pidstore(pid_value, object_uuid)
-            if texkey:
-                return
-            return super().create(
-                pid_value=pid_value,
-                status=PIDStatus.REGISTERED,
-                object_type=object_type,
-                object_uuid=object_uuid,
-                **kwargs,
-            )
-        else:
-            first_part = cls.build_texkey_first_part(data)
-            second_part = cls.build_texkey_second_part(data)
-            if first_part and second_part:
-                new_texkey = f"{first_part}:{second_part}"
-                texkeys_from_metadata = data.get("texkeys", [])
-                if len(texkeys_from_metadata) == 0 or not texkeys_from_metadata[
-                    0
-                ].startswith(new_texkey):
-                    return super().create(
-                        pid_value=cls.get_texkey_with_random_part(new_texkey),
-                        status=PIDStatus.REGISTERED,
-                        object_type=object_type,
+            if not texkey:
+                return super().create(
+                    pid_value=pid_value,
+                    status=PIDStatus.REGISTERED,
+                    object_type=object_type,
+                    object_uuid=object_uuid,
+                    **kwargs,
+                )
+            return
+
+        first_part = cls.build_texkey_first_part(data)
+        if not first_part:
+            raise TexkeyCannotGenerateFirstPart
```

Comment:
doesn't `build_texkey_first_part` already raise? if you're afraid you could get empty content there and want to check here, better remove the exception from that method.

Reply:
:+1: